tfx/experimental/pipeline_testing/stub_component_launcher.py

```diff
@@ -0,0 +1,81 @@
+# Lint as: python2, python3
+# Copyright 2020 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""stub component launcher which launches component executors and stub executors in process."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+from typing import Any, Dict, List, Text, Type
+
+from tfx import types
+from tfx.components.base import base_executor
+from tfx.experimental.pipeline_testing import base_stub_executor
+from tfx.orchestration.launcher import in_process_component_launcher
+
+class StubComponentLauncher(
+    in_process_component_launcher.InProcessComponentLauncher):
+  """Responsible for launching a stub executor.
+
+  The executor will be launched in the same process of the rest of the
+  component, i.e. its driver and publisher.
+  """
+
+  def _run_executor(self, execution_id: int,
+                    input_dict: Dict[Text, List[types.Artifact]],
+                    output_dict: Dict[Text, List[types.Artifact]],
+                    exec_properties: Dict[Text, Any]) -> None:
+    """Execute underlying component implementation."""
+    component_id = self._component_info.component_id
+    if component_id not in self.component_map:
+      super(StubComponentLauncher, self)._run_executor(execution_id,
+                                                       input_dict,
+                                                       output_dict,
+                                                       exec_properties)
+    else:
+      executor_context = base_executor.BaseExecutor.Context(
+          beam_pipeline_args=self._beam_pipeline_args,
+          tmp_dir=os.path.join(self._pipeline_info.pipeline_root, '.temp', ''),
+          unique_id=str(execution_id))
+      executor = self.component_map[component_id](component_id,
+                                                  self.test_data_dir,
+                                                  executor_context)
+      executor.Do(input_dict, output_dict, exec_properties)
+
+def create_stub_launcher_class(
```

Comment:
It is unclear why we should use a factory method instead of a normal __init__ function on the class. Maybe explain that in code comment?

Reply:
Thank you for the comment! But I think that it is not clear yet why we need a factory method. We might need to describe why we need to use a class rather than an instance of a class.